### PR TITLE
split addon gui for skin and binary addons

### PR DESCRIPTION
**Pull request by FernetMenta**
see title

@MartijnKaijser @opdenkamp follow-up from our discussion

### CODE REVIEW

File: addons/kodi.guilib/addon.xml.in [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<addon id="kodi.guilib" version="@guilib_version@" provider-name="Team-Kodi">
+  <backwards-compatibility abi="@guilib_version@"/>
```
Comment (MartijnKaijser) [inline addons/kodi.guilib/addon.xml.in]: shouldn't these be split between  KODI_GUILIB_API_VERSION and KODI_GUILIB_MIN_API_VERSION ?
i think the generated addon.xml also needs to be added to .gitignore

Comment (FernetMenta) [inline addons/kodi.guilib/addon.xml.in]: ok, will go another round with powershell, ugh :)

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
@FernetMenta same could be one for the PVR API/ABI versions to make it easier (other PR of course)

**Comment by opdenkamp** [comment]
looking good, +1
what do you mean @MartijnKaijser? the pvr one doesn't try to match to an api version used by python/skins?

**Comment by FernetMenta** [comment]
@opdenkamp he means the auto-generated version in addon.xml

**Comment by opdenkamp** [comment]
yeah found that out on irc, nice :)

**Comment by notspiff** [comment]
can you rationalize why you build these as separate objects ?

**Comment by opdenkamp** [comment]
they were already built as separate objects because guilib callbacks are not just for pvr add-ons, or am I misunderstanding the question?
the only difference is that the name changed and that we no longer try to match to the python API version used for skinning, which had nothing to do with these callbacks

**Comment by notspiff** [comment]
i know. it just gave a context..

they were built as separate objects to keep executable size down on xbox since it loaded the entire binary into memory. there's is absolutely no rationale for keeping them as separate objects these days.

**Comment by opdenkamp** [comment]
right okay, guess that's something to look into after all this add-on stuff settled down ;-)

**Comment by notspiff** [comment]
for reference; this is a q&d way to change it: https://github.com/notspiff/kodi-cmake/commit/1a6241b57c277778834ca1675934a9a0a4cb643e

**Comment by MartijnKaijser** [comment]
needs a rebase now.

**Comment by FernetMenta** [comment]
@opdenkamp can you merge this with your API changes? then we need to adjust the pvr addons only once

**Comment by opdenkamp** [comment]
we can press the button on both PRs within a few minutes as soon as jenkins is fixed to not fail on android. no need to sync these as it's not bumping the pvr api version, just the gui one, right?

**Comment by FernetMenta** [comment]
> we can press the button on both PRs within a few minutes

yes, that is what I suggested

> no need to sync these as it's not bumping the pvr api version, just the gui one, right?

it does not touch pvr API. I thought to remember you saying that pvr API needs a bump if any other APi it uses changed but I may be wrong.

**Comment by opdenkamp** [comment]
no, just the versions of the add-ons need to be bumped

**Comment by opdenkamp** [comment]
waiting for jenkins to finish building https://github.com/xbmc/xbmc/pull/6566 too, jenkins build this please
will pull them both if it succeeds

**Pull request merged**